.github/workflows/codeql.yml

```diff
@@ -2,11 +2,16 @@ name: "CodeQL"
 
 on:
   push:
-    branches: [main]
+    branches:
+      - '**'
 
   pull_request:
     branches: [main]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref || github.run_id }}
```

Comment:
what happens when it's not a pull request? `github.event.pull_request` is null, will this NPE trying to access `.number`?

Reply:
I understand OR, my concern is whether it will throw an exception on non-PR because it cannot evaluate the first token (null pointer exception)

Reply:
No. It won't give an exception like that. I have tested it with push events also, where the `github.event.pull_request.number` is expected to be null or undefined.

Reply:
to put this differently, if this was javascript running on non-PR, `github.event.pull_request.number` would bomb, but `(github.event.pull_request && github.event.pull_request.number)` would always execute.

Reply:
Official docs explain it better: https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value

Reply:
> to put this differently, if this was javascript running on non-PR, github.event.pull_request.number would bomb, but (github.event.pull_request && github.event.pull_request.number) would always execute.

Got your concern, but it is different from JS, though your syntax is also correct, and can be used to be double sure

Reply:
Updated the syntax 🚀